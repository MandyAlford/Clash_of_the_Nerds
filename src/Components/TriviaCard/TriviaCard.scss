.trivia-card {
  background-color: rgba(255, 255, 255, 0.9);
  padding: 30px;
  border: 5px solid #662324;
  align-items: center;
  height: 300px;
  width: 400px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.trivia-question {
  background-color: rgba(255, 255, 255, 0.9);
  padding: 30px;
  border: 5px solid #662324;
}

.trivia-choice {
  width: 200px;
  height: 50px;
  color: #662324;
  font-size: 20px;
  margin: 5px;
  border-radius: 15px;
}

.trivia-choice:hover {
  background-color: #662324;
  color: #FFF;
}

.results-route {
  width: 200px;
  height: 50px;
  background-color: #662324;
  color: #FFF;
  font-size: 20px;
  margin: 5px;
  border-radius: 15px;
}
