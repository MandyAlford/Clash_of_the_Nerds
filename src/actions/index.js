export const addUser = (user) => ({
    type: 'ADD_USER',
    user
})

export const addTrivia = (trivia) => ({
    type: 'ADD_TRIVIA',
    trivia
})

export const addCorrectGuess = (guess) => ({
  type: 'ADD_CORRECT_GUESS',
  guess
})

export const addIncorrectGuess = (guess) => ({
  type: 'ADD_INCORRECT_GUESS',
  guess
})

export const resetAnswers = () => ({
  type: 'RESET_STORE_ANSWERS'
})

export const saveUsersTrivia = (trivia) => ({
  type: 'SAVE_TRIVIA',
  trivia
})
