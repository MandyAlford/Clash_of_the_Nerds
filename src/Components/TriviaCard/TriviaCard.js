import React, { Component } from 'react';
import './TriviaCard.scss';
import { connect } from 'react-redux';
import { addIncorrectGuess, addCorrectGuess } from '../../actions';


// category: "Mythology"
// type: "multiple"
// difficulty: "easy"
// question: "The greek god Poseidon was the god of what?"
// correct_answer: "The Sea"
// incorrect_answers: (3) ["War", "Sun", "Fire"]

class TriviaCard extends Component{
  constructor(props) {
    super(props)
    this.state = {
      choice: ''
    }
  }
  updateChoice = (event) => {
    if(event.target.value === this.props.trivia.correct_answer) {
      this.props.loadCorrectGuessToStore(event.target.value)
    } else {
      this.props.loadIncorrectGuessToStore(this.props.trivia)
      this.props.loadIncorrectGuessToStore(event.target.value)
    }
    this.setState({ choice: event.target.value })
    this.props.changeCard(this.props.trivia)
  }
  render() {
    let allAnswers = this.props.trivia.incorrect_answers.concat(this.props.trivia.correct_answer)
    let formattedAnswers = allAnswers.map(answer => {
      return <button type='button' className='trivia-choice' value={answer} key={Math.random()} onClick={this.updateChoice}>{answer}</button>
    })
    return (
      <div className='trivia-card'>
      <div>{formattedAnswers}</div>
      </div>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  loadCorrectGuessToStore: (guess) => { dispatch(addCorrectGuess(guess)) },
  loadIncorrectGuessToStore: (guess) => { dispatch(addIncorrectGuess(guess)) }
})

export default connect(null, mapDispatchToProps)(TriviaCard)
