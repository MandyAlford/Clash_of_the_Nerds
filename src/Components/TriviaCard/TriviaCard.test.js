import React from 'react';
import { TriviaCard, mapDispatchToProps } from './TriviaCard';
import { shallow } from 'enzyme';
import { mockRandom } from 'jest-mock-random';
import { addCorrectGuess, addIncorrectGuess } from '../../actions';

describe('TriviaCard', () => {
  let wrapper, trivia, loadCorrectGuessToStoreMock, loadIncorrectGuessToStoreMock;

  beforeEach(() => {
    mockRandom([0.1]);
    trivia={
      category: 'Mythology',
      type: 'multiple',
      difficulty: 'medium',
      question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
      correct_answer: 'Athena',
      incorrect_answers: ['Ares','Artemis','Apollo'],
      userGuess: 'Athena'
    };
    loadCorrectGuessToStoreMock = jest.fn().mockImplementation();
    loadIncorrectGuessToStoreMock = jest.fn().mockImplementation();
    wrapper = shallow(<TriviaCard
      trivia={trivia}
      changeCard={jest.fn()}
      loadCorrectGuessToStore={loadCorrectGuessToStoreMock}
      loadIncorrectGuessToStore={loadIncorrectGuessToStoreMock}
      />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update correct trivia question when updateChoice is called', () => {
    let mockEvent = {
      target: {
        name: 'title',
        value: 'Athena'
      }
    }
    let mockResponse = {
      category: 'Mythology',
      type: 'multiple',
      difficulty: 'medium',
      question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
      correct_answer: 'Athena',
      incorrect_answers: ['Ares','Artemis','Apollo'],
      userGuess: 'Correct!'
    }
    wrapper.instance().updateChoice(mockEvent)
    expect(loadCorrectGuessToStoreMock).toHaveBeenCalledWith(mockResponse)
  })

  it('should shuffle answers in array before displaying them to user', () => {
    let mockAnswers = ['Ares', 'Artemis', 'Apollo', 'Athena']
    wrapper.instance().shuffle(mockAnswers)
    let expected = ['Athena', 'Apollo', 'Artemis', 'Ares']
    expect(mockAnswers).toEqual(expected);
  })

  it('should update incorrect trivia question when updateChoice is called', () => {
    let mockEvent = {
      target: {
        name: 'title',
        value: 'Apollo'
      }
    }
    let mockResponse = {
      category: 'Mythology',
      type: 'multiple',
      difficulty: 'medium',
      question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
      correct_answer: 'Athena',
      incorrect_answers: ['Ares','Artemis','Apollo'],
      userGuess: 'Apollo'
    }
    wrapper.instance().updateChoice(mockEvent)
    expect(loadIncorrectGuessToStoreMock).toHaveBeenCalledWith(mockResponse)
  })

  it('should call dispatch with the addCorrectGuess action when loadCorrectGuessToStore is called', () => {
    const mockDispatch = jest.fn();
    const trivia = {
      category: 'Mythology',
      type: 'multiple',
      difficulty: 'medium',
      question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
      correct_answer: 'Athena',
      incorrect_answers: ['Ares','Artemis','Apollo'],
      userGuess: 'Athena'
    }
    const actionToDispatch = addCorrectGuess(trivia)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.loadCorrectGuessToStore(trivia)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('should call dispatch with the addIncorrectGuess action when loadCorrectGuessToStore is called', () => {
    const mockDispatch = jest.fn();
    const trivia = {
      category: 'Mythology',
      type: 'multiple',
      difficulty: 'medium',
      question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
      correct_answer: 'Athena',
      incorrect_answers: ['Ares','Artemis','Apollo'],
      userGuess: 'Athena'
    }
    const actionToDispatch = addIncorrectGuess(trivia)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.loadIncorrectGuessToStore(trivia)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})
