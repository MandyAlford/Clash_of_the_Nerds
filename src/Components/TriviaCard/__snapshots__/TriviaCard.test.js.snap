// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriviaCard should match snapshot 1`] = `
<div
  className="trivia-card"
>
  <div>
    <button
      className="trivia-choice"
      key="0.1"
      onClick={[Function]}
      type="button"
      value="Athena"
    >
      Athena
    </button>
    <button
      className="trivia-choice"
      key="0.1"
      onClick={[Function]}
      type="button"
      value="Apollo"
    >
      Apollo
    </button>
    <button
      className="trivia-choice"
      key="0.1"
      onClick={[Function]}
      type="button"
      value="Artemis"
    >
      Artemis
    </button>
    <button
      className="trivia-choice"
      key="0.1"
      onClick={[Function]}
      type="button"
      value="Ares"
    >
      Ares
    </button>
  </div>
</div>
`;
